python/ray/util/spark/cluster_init.py

```diff
@@ -1232,6 +1179,216 @@ def setup_ray_cluster(
+        "remote_connection_address" is in format of
+        "ray://<ray_head_node_ip>:<ray-client-server-port>",
+        if your client runs on a machine that also hosts a Ray cluster node locally,
+        you can connect to the Ray cluster via ``ray.init(address)``,
+        otherwise you can connect to the Ray cluster via
+        ``ray.init(remote_connection_address)``.
+    """
+
+    return _setup_ray_cluster_internal(
+        num_worker_nodes=num_worker_nodes,
+        num_cpus_worker_node=num_cpus_worker_node,
+        num_cpus_head_node=num_cpus_head_node,
+        num_gpus_worker_node=num_gpus_worker_node,
+        num_gpus_head_node=num_gpus_head_node,
+        object_store_memory_worker_node=object_store_memory_worker_node,
+        object_store_memory_head_node=object_store_memory_head_node,
+        head_node_options=head_node_options,
+        worker_node_options=worker_node_options,
+        ray_temp_root_dir=ray_temp_root_dir,
+        strict_mode=strict_mode,
+        collect_log_to_path=collect_log_to_path,
+        autoscale=autoscale,
+        autoscale_upscaling_speed=autoscale_upscaling_speed,
+        autoscale_idle_timeout_minutes=autoscale_idle_timeout_minutes,
+        is_global=False,
+        **kwargs,
+    )
+
+
+def setup_global_ray_cluster(*args, **kwargs):
+    """
+    Set up a global mode cluster.
+    The global Ray on spark cluster means:
+    - You can only create one active global Ray on spark cluster at a time.
+    On databricks cluster, the global Ray cluster can be used by all users,
+    - as contrast, non-global Ray cluster can only be used by current notebook
+    user.
+    - It is up persistently without automatic shutdown.
+    - On databricks notebook, you can connect to the global cluster by calling
+    ``ray.init()`` without specifying its address, it will discover the
+    global cluster automatically if it is up.
+
+    For global mode, the ``ray_temp_root_dir`` argument is not supported.
+    Global model Ray cluster always use the default Ray temporary directory
+    path.
+
+    All arguments are the same with ``setup_ray_cluster`` API except that:
+    - the ``ray_temp_root_dir`` argument is not supported.
+    Global model Ray cluster always use the default Ray temporary directory
+    path.
+    - A new argument "is_blocking" (default ``True``) is added.
+    If "is_blocking" is True,
+    then keep the call blocking until it is interrupted.
+    once the call is interrupted, the global Ray on spark cluster is shut down and
+    `serve_global_ray_cluster` call terminates.
+    If "is_blocking" is False,
+    once Ray cluster setup completes, return immediately.
+    """
+
+    if "ray_temp_root_dir" in kwargs:
+        raise ValueError("'ray_temp_root_dir' argument is not supported in global mode")
+
+    if "is_global" in kwargs:
+        raise ValueError(
+            "You can't set 'is_global' argument in 'serve_global_ray_cluster' call."
+        )
```

Comment:
These won't happen if we explicitly list out all the parameters supported by `setup_global_ray_cluster`.